text/0000-safety-tags.md

```diff
@@ -0,0 +1,481 @@
+- Feature Name: safety_tag
+- Start Date: 2025-07-29
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC introduces a concise safety-comment convention for unsafe code in libstd-adjacent crates:
+tag every unsafe function and call with `#[safety { SP1, SP2 }]`.
+
+Safety tags refine today’s safety-comment habits: a featherweight syntax that condenses every
+requirement into a single, check-off reminder.
+
+The following snippet [compiles] today, but we expect Clippy and Rust-Analyzer to enforce tag checks
+and provide first-class IDE support.
+
+[compiles]: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=056cbe626a7cc92a317e38e9f54cb1fb
+
+```rust
+#![feature(custom_inner_attributes)]
+#![clippy::safety(invariant::ValidPtr)] // 💡
+
+pub mod invariant {
+    #[clippy::safety::tag]
+    pub fn ValidPtr() {}
+}
+
+#[clippy::safety { ValidPtr }] // 💡
+pub unsafe fn read<T>(ptr: *const T) {}
```

Comment:
What about functions that take several parameters with some different, some overlapping requirements? Would it be possible to scope some tags to just some arguments (or maybe even type generics) to require the caller to discharge the requirements separately for each?